tests/baselines/reference/keyofAndIndexedAccessErrors.js

```diff
@@ -142,6 +142,16 @@ function test1<T extends Record<string, any>, K extends keyof T>(t: T, k: K) {
     t[k] = "hello";  // Error
     t[k] = [10, 20];  // Error
 }
+
+// Repro from #28839
+
+function f30<T, K extends keyof T>() {
+    let x: Partial<Record<keyof T, string>>[K] = "hello";
+}
+
+function f31<T, K extends keyof T>() {
+    let x: Partial<Partial<Partial<Partial<Partial<Partial<Partial<Record<keyof T, string>>>>>>>>[K] = "hello";
```

Comment:
Aside: Part of me wonders if we should have come rule in place to compact mapped types like this, even when generic.